src/ILCompiler.RyuJit/src/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -130,6 +130,7 @@ private void ComputeLookup(ref CORINFO_RESOLVED_TOKEN pResolvedToken, object ent
                     {
                         lookup.runtimeLookup.offset1 = IntPtr.Zero;
                     }
+                    lookup.runtimeLookup.sizeOffset = CORINFO.CORINFO_NO_SIZE_CHECK;
```

Comment:
RyuJIT sometimes tried to emit the dictionary size check due to this that made it to go side-ways.